### PR TITLE
Capability to process images as part of multipart message body

### PR DESCRIPTION
**Pull request by bbouffaut**
Add the capability to process images embedded into multipart messages body:
- add a multipart message type test in current get_image function
- put current get_image function content into return_image function
- return_image function is called either image is in attachments or in body parts

NB:
- the current version only process JPEG image
- in case of multiple images in body, only retrieve the first one

### CODE REVIEW

**Comment by jamesboswell** [comment]
Merci @bbouffaut   I will try to review in the next few days, I have a lot going on right now.  soyez patient s'il vous plaît

File: thold-slack.py [hunk 1]
```diff
@@ -190,23 +190,39 @@ def get_body(message):
...
 
 def get_image(message):
-    """Get the image attachment"""
+    """UPDATE BBO: Check if message is multipart or not. 
+    If so, then parse message body and extract image (JPEG only in this version,
+    but it would be quite easy to loop over any kinds of image types"""
 
-    try:
+    if message.is_multipart():
+        #print("Message is multipart") #DEBUG
```
Comment (jamesboswell) [inline thold-slack.py:205]: @bbouffaut can we remove the DEBUG lines please?

File: thold-slack.py [hunk 2]
```diff
@@ -190,23 +190,39 @@ def get_body(message):
...
+        #get the FIRST image JPEG version only
+        image_parts = [part for part in typed_subpart_iterator(message,'image','jpg')]
+        image_part = image_parts[0]
+        return return_image(image_part)
+
+    else:
+        #print("Message is not multipart but has attachement") #DEBUG
```
Comment (jamesboswell) [inline thold-slack.py:212]: Remove DEBUG

File: thold-slack.py [hunk 3]
```diff
@@ -190,23 +190,39 @@ def get_body(message):
...
         attachment = message.get_payload()[1]
+        return return_image(attachment)
+    
+
+def return_image(part_or_attachement):
+    '''UPDATE BBO: extract the image data from the part provided as argument: either part of multipart message, either attachement'''
```
Comment (jamesboswell) [inline thold-slack.py:218]: Remove BBO please
```
'''Extract the image data from the part provided as argument: either part of multipart message, either attachment'''```

File: thold-slack.py [hunk 4]
```diff
@@ -190,23 +190,39 @@ def get_body(message):
...
+
+    try:
 
-        if 'image/jpg' in attachment.get_content_type():
-            path = image_path
-	    imgfilename = genfile()
-            imgfile = path + imgfilename
-            open(imgfile, 'wb').write(attachment.get_payload(decode=True))
+        if 'image/jpg' in part_or_attachement.get_content_type():
```
Comment (jamesboswell) [inline thold-slack.py:222]: Anglais form  `attachment` s'il vous plait

File: thold-slack.py [hunk 5]
```diff
@@ -190,23 +190,39 @@ def get_body(message):
...
 
-        if 'image/jpg' in attachment.get_content_type():
-            path = image_path
-	    imgfilename = genfile()
-            imgfile = path + imgfilename
-            open(imgfile, 'wb').write(attachment.get_payload(decode=True))
+        if 'image/jpg' in part_or_attachement.get_content_type():            
+            imgfilename = genfile('jpg')
+            imgfile = image_path + imgfilename
+            open(imgfile, 'wb').write(part_or_attachement.get_payload(decode=True))
```
Comment (jamesboswell) [inline thold-slack.py:225]: `attachment`

**Review by jamesboswell — CHANGES_REQUESTED** [review]
This looks pretty neat!  Thank you @bbouffaut   Can you please make the suggested changes?
I need to do a bit of testing, and then I can accept.

**Comment by bbouffaut** [comment]
Hi @jamesboswell ! Fix is done, and update is pushed. Thanks for your feedback.

**Pull request merged**